plugins/env/env.go

```diff
@@ -0,0 +1,59 @@
+// This plugins creates the environment variables setup in the payload
+// env" section. It does its processing at BuildSandbox call. After the
+// "BuildSandox" stage, all environment variables configured in the payload
+// are available in the SandboxBuilder.
+
+package env
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/plugins"
+	"github.com/taskcluster/taskcluster-worker/plugins/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+type plugin struct {
+	plugins.PluginBase
+}
+
+func (plugin) PayloadSchema() (runtime.CompositeSchema, error) {
+	return envPayloadSchema, nil
+}
+
+func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlugin, error) {
+	return taskPlugin{
+		TaskPluginBase: plugins.TaskPluginBase{},
+		payload:        options.Payload.(envPayload),
+	}, nil
+}
+
+type taskPlugin struct {
+	plugins.TaskPluginBase
+	payload envPayload
+}
+
+func (self taskPlugin) BuildSandbox(sandboxBuilder engines.SandboxBuilder) error {
+	for k, v := range self.payload {
+		err := sandboxBuilder.SetEnvironmentVariable(k, v)
+		if err != nil {
+			return err
```

Comment:
Most errors here should be transformed to `MalformedPayloadError` see: https://github.com/taskcluster/taskcluster-worker/blob/2130156c8e55bdc710af9a7c16e5877844ef7f17/engines/sandboxbuilder.go#L64-L73

Basically, `ErrNamingConflict` is a problem that can be solved by changing the task payload. The same is the case for `ErrFeatureNotSupported`, if env vars aren't support... user just shouldn't specify any...